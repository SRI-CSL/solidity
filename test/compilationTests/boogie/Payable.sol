pragma solidity ^0.4.23;

contract PayableFunctions {

    function receive(uint param) public payable returns (uint) {
        return param + msg.value;
    }
}

contract Other {
    PayableFunctions p;

    function transfer(uint amount) public returns (uint) {
        return p.receive.gas(1).value(amount)(100);
    }

    function transferNested(uint amount) public returns (uint) {
        return p.receive.value(p.receive.value(100)(200))(p.receive.value(amount)(300));
    }

    function __verifier_main() public {
        assert(transfer(1) == 101);
        assert(transferNested(1) == 601);
    }
}
