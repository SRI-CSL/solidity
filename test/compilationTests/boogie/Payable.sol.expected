Verifying PayableFunctions::receive: OK
Verifying Payable::__verifier_main: OK
Verifying Payable::transferNested: ERROR
 - Source Payable.sol, line 31, col 55: Calling payable function might fail due to insufficient ether
 - Source Payable.sol, line 31, col 16: Calling payable function might fail due to insufficient ether
